systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -348,4 +352,71 @@ void testCruiseControlIntraBrokerBalancingWithoutSpecifyingJBODStorage(Extension
         KafkaRebalanceStatus kafkaRebalanceStatus = KafkaRebalanceResource.kafkaRebalanceClient().inNamespace(testStorage.getNamespaceName()).withName(testStorage.getClusterName()).get().getStatus();
         assertThat(kafkaRebalanceStatus.getConditions().get(0).getReason(), is("InvalidResourceException"));
     }
+
+    @ParallelNamespaceTest
+    void testCruiseControlDuringBrokerScaleUpAndDown(ExtensionContext extensionContext) {
+        final TestStorage testStorage = new TestStorage(extensionContext);
+        final int initialReplicas = 3;
+        final int scaleTo = 5;
+
+        resourceManager.createResource(extensionContext,
+            KafkaTemplates.kafkaWithCruiseControl(testStorage.getClusterName(), initialReplicas, initialReplicas).build(),
+            KafkaTopicTemplates.topic(testStorage.getClusterName(), testStorage.getTopicName(), 10, 3).build()
+        );
+
+        Map<String, String> kafkaPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getKafkaSelector());
+
+        LOGGER.info("Scaling Kafka up to {}", scaleTo);
+
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), kafka -> kafka.getSpec().getKafka().setReplicas(scaleTo), testStorage.getNamespaceName());
+        kafkaPods = RollingUpdateUtils.waitForComponentScaleUpOrDown(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), scaleTo, kafkaPods);
+
+        LOGGER.info("Checking that {} topic has replicas on first 3 brokers", testStorage.getTopicName());
+        List<String> topicReplicas = KafkaTopicUtils.getKafkaTopicReplicasForEachPartition(testStorage.getNamespaceName(), testStorage.getTopicName(), KafkaResources.kafkaPodName(testStorage.getClusterName(), 0), KafkaResources.plainBootstrapAddress(testStorage.getClusterName()));
+        assertEquals(0, (int) topicReplicas.stream().filter(line -> line.contains("3") || line.contains("4")).count());
```

Comment:
Let's add this test even before the scaling up when the topic was created.

Reply:
Makes sense, done :)